setup-environment-internal

```diff
@@ -247,10 +247,12 @@ ln -sf "${MANIFESTS}" "${OEROOT}"/layers/
 
 DISTRO_DIRNAME=$(echo "${DISTRO}" | sed 's#[.-]#_#g')
 
-LMP_VERSION_CACHE="${LMP_VERSION_CACHE:-$(git --git-dir ${MANIFESTS}/.git describe HEAD --tags --abbrev=0)}"
-if [ -v LMP_VERSION_CACHE_DEV ]; then
-    # to use the development version of the cache the user need to define the LMP_VER_DEV env
-    LMP_VERSION_CACHE=$(( $LMP_VERSION_CACHE + 1 ))
+if ! [ -n "$LMP_VERSION_CACHE" ]; then
```

Comment:
You can use -z here instead of ! -n.

Reply:
I have read somewhere that -z does not correctly interpret empty strings but after testing I couldn't confirm it, so will change it to `-z`.

Reply:
Hm, weird, that is a first for me at least :-)

```
       -z string
              True if the length of string is zero.
       string
       -n string
              True if the length of string is non-zero.
```

They should behave "similarly".

Reply:
yeah, one is the opositor of the other. the two methods works in the same way, even with the variable undefined `unset LMP_VERSION_CACHE`